src/main/java/com/minecolonies/blocks/AbstractBlockHut.java

```diff
@@ -77,23 +76,75 @@ private void initBlock()
 
     @NotNull
     @Override
-    public TileEntity createNewTileEntity(World world, int meta)
+    public TileEntity createNewTileEntity(final World world, final int meta)
     {
         //Creates a tile entity for our building
         return new TileEntityColonyBuilding();
     }
 
     @Override
-    public int getMetaFromState(@NotNull IBlockState state)
+    public boolean isFullBlock(final IBlockState state)
     {
-        return state.getValue(FACING).getIndex();
+        return false;
     }
 
+    /**
+     * Convert the given metadata into a BlockState for this Block
+     */
+    @NotNull
+    @Override
+    public IBlockState getStateFromMeta(final int meta)
+    {
+        EnumFacing enumfacing = EnumFacing.getFront(meta);
+
+        if (enumfacing.getAxis() == EnumFacing.Axis.Y)
+        {
+            enumfacing = EnumFacing.NORTH;
+        }
+
+        return this.getDefaultState().withProperty(FACING, enumfacing);
+    }
 
     // =======================================================================
     // ======================= Rendering & IBlockState =======================
     // =======================================================================
 
+    /**
+     * Convert the BlockState into the correct metadata value
+     */
+    @Override
+    public int getMetaFromState(final IBlockState state)
+    {
+        return state.getValue(FACING).getIndex();
+    }
+
+    //We unfortunately have to implement these two, to rotate our blocks in the structures.
+    @NotNull
+    @Override
+    public IBlockState withRotation(@NotNull final IBlockState state, final Rotation rot)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Don't override a deprecated method or explicitly mark it as "@Deprecated". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)